openedx/core/djangoapps/api_admin/api/v1/views.py

```diff
@@ -0,0 +1,78 @@
+"""
+API Views.
+"""
+
+from django.contrib.auth.models import User
+from django.utils.translation import ugettext_lazy as _
+
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework.authentication import SessionAuthentication
+from rest_framework.exceptions import NotFound, PermissionDenied
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.generics import ListAPIView
+from rest_framework_oauth.authentication import OAuth2Authentication
+
+from openedx.core.djangoapps.api_admin.api.v1 import serializers as api_access_serializers
+from openedx.core.djangoapps.api_admin.models import ApiAccessRequest
+
+
+class ApiAccessRequestView(ListAPIView):
+    """
+    Return `API Access Requests` in the form of a paginated list.
+
+    Example:
+        `GET: /api-admin/api/v1/api_access_request/`
+        {
+            "count": 1,
+            "num_pages": 1,
+            "current_page": 1,
+            "results": [
+                {
+                    "id": 1,
+                    "created": "2017-09-25T08:41:48.934364Z",
+                    "modified": "2017-09-25T08:42:04.185209Z",
+                    "user": 6,
+                    "status": "denied",
+                    "website": "https://www.example.com/",
+                    "reason": "Example",
+                    "company_name": "Example Name",
+                    "company_address": "Silicon Valley",
+                    "site": 1,
+                    "contacted": true
+                }
+            ],
+            "next": null,
+            "start": 0,
+            "previous": null
+        }
+    """
+    authentication_classes = (JwtAuthentication, OAuth2Authentication, SessionAuthentication,)
+    permission_classes = (IsAuthenticated, )
+    serializer_class = api_access_serializers.ApiAccessRequestSerializer
+
+    def get_queryset(self):
+        username = self.request.query_params.get('username', None)
+        queryset = ApiAccessRequest.objects.all()
+
+        # Filter based on given username, if requesting user has staff access.
+        if self.request.user.is_staff:
+            if username:
+                try:
+                    user = User.objects.get(username=username)
+                except User.DoesNotExist:
+                    raise NotFound(_('No user with the username [{username}] exists.').format(username=username))
+
+                queryset = queryset.filter(user=user)
+
+        # Raise `PermissionDenied` exception if requesting user does not have
+        # access and is asking to get filtered records based on username.
+        elif username:
+            raise PermissionDenied(
```

Comment:
nit: Please add this `PermissionDenied` exception and the exception `NotFound` above to the `Raises:` section of the docstring.

Reply:
Done